### ISSUE TITLE
Adobe Comment on 3.3.7 Accessible Authentication

### ISSUE THREAD

**Issue by awkawk** [w3c/wcag#1885]
Adobe supports this proposed SC, but wants to highlight one concern and one question: 

The concern is that this SC will place a higher burden on small organizations who are likely to be less familiar with security concerns. Most people (even people in the tech field) are not fully aware of all the nuances, threats, and recent updates related to Security, and this SC will require a higher level of engagement by these people. 

The question is whether the WG anticipates any problem with sites conforming to Timing Adjustable with this SC as there may be time limits associated with the WebAuthn processes.

**Comment by alastc** [comment]
> The concern is that this SC will place a higher burden on small organizations who are likely to be less familiar with security concerns.

That is something I had considered as I work with a couple of smaller organisations, even running a site for one.  Not that I know everything about it, but I think it really depends on the tools used to create the site. For example various CMSs like Wordpress and Drupal, or the framework used. 

The tools provide the login function, or at least the basic form of it, therefore it will come down to those projects meeting the criteria, or providing modules (such as WebAuthn modules) to meet it. 

Even creating a _reasonable_ password storage function from scratch is (or should be!) out of scope for a individual or small dev team, considering how easy it is to get wrong.

>  any problem with sites conforming to Timing Adjustable with this SC as there may be time limits associated with the WebAuthn processes

Interesting question, I assume that if you have both username/password and WebAuthn  as mechanisms you'd pass as there is an alternative? (See #1879)

**Comment by alastc** [comment]
Proposed response:

<hr>

The basic case of a username/password field can pass with reasonable markup, and for small organisations that is generally what is provided by CMSs or frameworks. Therefore the burden should not increase.

For the time-out during authentication with WebauthN ([spec link](https://www.w3.org/TR/webauthn-2/#ref-for-dom-publickeycredentialcreationoptions-timeout)), that would fit the "essential exception" of [2.2.1](https://www.w3.org/TR/WCAG22/#timing-adjustable), as the time-out is required to maintain the security of the process.

**Comment by alastc** [comment]
Needs someone to check how WebauthN impacts 2.2.1 timing adjustable.

**Comment by alastc** [comment]
Updated response:

<hr>

The basic case of a username/password field can pass with reasonable markup, and for small organisations that is generally what is provided by CMSs or frameworks. Therefore the burden should not increase.

For the time-out during authentication with WebauthN ([spec link](https://www.w3.org/TR/webauthn-2/#ref-for-dom-publickeycredentialcreationoptions-timeout)), the time-out does not loose anything, you simply re-start the authentication, it is no more of a step than if there were a pop-over warning you of the time out.

**Comment by awkawk** [comment]
@alastc So are we saying that restarting a process where there is a timeout is sufficient? I'm just wondering what part of 2.2.1 we would point to as being supported in this process...

**Comment by awkawk** [comment]
@alastc I see you indicated that it is related to "essential", but we have not used "essential" to be security-related as the definition doesn't really fit. That's why 3.3.8 treats these separately.

**Comment by alastc** [comment]
I'm saying it is a form of infinite extensions. It does time out, but there is no difference between a pop-up saying you need to extend, and just starting again.
